connect/runtime/src/main/java/org/apache/kafka/connect/runtime/HerderConnectorContext.java

```diff
@@ -38,4 +38,9 @@ public void requestTaskReconfiguration() {
         // Distributed herder will forward the request to the leader if needed
         herder.requestTaskReconfiguration(connectorName);
     }
+
+    @Override
+    public void raiseError(Exception e) {
+        herder.onFailure(connectorName, e);
```

Comment:
I think there's a thread safety issue here. This will be called from a connector thread, but I don't think any of the relevant methods are synchronized. The AbstractHerder will forward the request to the backing store, but neither backing store seems to do any synchronization. Other related methods (e.g. requestTaskReconfiguration) are just synchronized at the Herder level to avoid this problem.